lib/OpenStackExtract/MiqOpenStackVm/MiqOpenStackCommon.rb

```diff
@@ -50,6 +53,57 @@ def get_image_file_glance_v2(image_id)
     end
     tf
   end
+  
+  def create_image_from_snapshot(snapshot_id, disk_format)
+    snapshot = volume_service.snapshots.get(snapshot_id)
+    volume_options = {
+      :name        => "Temp Volume from #{snapshot.name}",
+      :size        => snapshot.size,
+      :snapshot_id => snapshot_id
+    }
+    volume = volume_service.volumes.new(volume_options)
+    volume.save
```

Comment:
@tzumainn 
If the `volume.save` raises an error, does it mean the volume doesn't need to be destroyed?
If not, then the `volume.save` should be moved inside the `begin` block.